core/src/test/java/org/apache/iceberg/hadoop/CommonCatalogTests.java

```diff
@@ -0,0 +1,443 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hadoop;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NamespaceNotEmptyException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.io.inmemory.InMemoryCatalog;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.transforms.Transform;
+import org.apache.iceberg.transforms.Transforms;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.iceberg.NullOrder.NULLS_FIRST;
+import static org.apache.iceberg.SortDirection.ASC;
+
+/**
+ * Common tests for the Iceberg catalog.
+ * The tests expect the catalog to support namespaces.
+ * <p>
+ * Currently, the following catalogs are tested:
+ * <ol>
+ *   <li>{@link HadoopCatalog}</li>
+ *   <li>{@link InMemoryCatalog}</li>
+ * </ol>
+ * Catalog specific tests should be tested in a separate test,
+ * e.g. {@link TestHadoopCatalog}.
+ */
+@RunWith(Parameterized.class)
+public class CommonCatalogTests extends HadoopTableTestBase {
+
+  private enum CatalogFactory {
```

Comment:
Is there a plan to use this test class to test other catalogs as well? Maybe these tests could be used to validate the Catalog implementations, so if the tests are passing then we can "certify" that the implementation adheres to the Iceberg Catalog standard.

I think it would be good to test HiveCatalog, GlueCatalog, JDBCCatalog etc, but I do not see a way to extend this ATM.
Do I miss something?

Reply:
+1 on that idea of having a way to "certify" catalog implementations